d3a_api_client/aggregator.py

```diff
@@ -0,0 +1,114 @@
+import logging
+
+from d3a_api_client.utils import logging_decorator, blocking_get_request, \
+    blocking_post_request
+from d3a_api_client.websocket_device import WebsocketMessageReceiver, WebsocketThread
+from concurrent.futures.thread import ThreadPoolExecutor
+from d3a_api_client.rest_device import RestDeviceClient
+from d3a_api_client.constants import MAX_WORKER_THREADS
+
+
+root_logger = logging.getLogger()
+root_logger.setLevel(logging.INFO)
+
+
+class AggregatorWebsocketMessageReceiver(WebsocketMessageReceiver):
+    def __init__(self, rest_client):
+        super().__init__(rest_client)
+
+    def received_message(self, message):
+        if "event" in message:
+            if message["event"] == "market":
+                self.client._on_market_cycle(message)
+            elif message["event"] == "tick":
+                self.client._on_tick(message)
+            elif message["event"] == "trade":
+                self.client._on_trade(message)
+            elif message["event"] == "finish":
+                self.client._on_finish(message)
+            elif message["event"] == "selected_by_device":
+                self.client._selected_by_device(message)
+            elif message["event"] == "unselected_by_device":
+                self.client._unselected_by_device(message)
+            else:
+                logging.error(f"Received message with unknown event type: {message}")
+        elif "command" in message:
+            self.command_response_buffer.append(message)
+
+
+class Aggregator(RestDeviceClient):
+
+    def __init__(self, simulation_id, domain_name, aggregator_name,
+                 websockets_domain_name, accept_all_devices=True):
+        super().__init__(
+            simulation_id=simulation_id, device_id="", domain_name=domain_name,
+            websockets_domain_name=websockets_domain_name, autoregister=False, start_websocket=False)
+
+        self.aggregator_name = aggregator_name
+        self.accept_all_devices = accept_all_devices
+        self.device_uuid_list = []
+        self.aggregator_uuid = None
+        user_aggrs = self.list_aggregators()
+        for a in user_aggrs:
+            if a["name"] == self.aggregator_name:
+                self.aggregator_uuid = a["uuid"]
+        if self.aggregator_uuid is None:
+            aggr = self._create_aggregator()
+            self.aggregator_uuid = aggr["uuid"]
+        self.start_websocket_connection()
```

Comment:
Minor but i would like it this to be refactored into its own function called "connect_to_simulation" or something like this. But skip it if you dont agree.

Reply:
Will do, thanks